### PR TITLE
Remove event cart message_template, tables from core

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
FK_civicrm_participant_cart_id

This started out as trying to deal with the message template sending me down a rabbit hole but I decided to try to move the tables too.

When enabled event cart functionality is available from the event info page

Before
----------------------------------------
Event cart is a hidden core extension - on most sites it will be disabled but the tables related to it will be in core as will the message template

After
----------------------------------------
tables & template moved to the extension

Technical Details
----------------------------------------
The tables move OK in my testing but it would be nice to remove cart_id from civicrm_participant too...

I removed the property from the DAO as those properties are there for autocomplete only & are not functional

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@colemanw what am I missing - I get this in the UI too on the event info page

```
TypeError: class@anonymous::getTableForEntity(): Return value must be of type string, null returned in class@anonymous/home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/mixin/lib/civimix-schema/src/SqlGenerator.php:12$80->getTableForEntity() (line 204 of /home/homer/buildkit/build/build-4/web/sites/all/modules/civicrm/mixin/lib/civimix-schema/src/SqlGenerator.php).
/home/homer/buildkit/app/c
```

**Comment by colemanw** [comment]
Hmm, I'm not sure but those install/uninstall.sql files shouldn't exist.

**Comment by colemanw** [comment]
@eileenmcnaughton I just pushed up a few fixes, but the underlying problem is going to persist: `Participant.cart_id` is a FK to a table that no longer exists (at least not in core) so the CiviCRM installer fails now because it can't create the foreign key. (that's why `getTableForEntity()` is causing an error, it can't find the entity).

**Comment by eileenmcnaughton** [comment]
@colemanw yeah - I'm trying to remove that FK & only add it when the cart is installed

**Comment by eileenmcnaughton** [comment]
OK - so deleting the install / uninstall lost

```
ALTER TABLE civicrm_participant ADD CONSTRAINT
  `FK_civicrm_participant_cart_id`
  FOREIGN KEY (`cart_id`)
    REFERENCES `civicrm_event_carts`(`ID`);
```

and 
```
ALTER TABLE civicrm_participant DROP CONSTRAINT FK_civicrm_participant_cart_id;
```

**Comment by colemanw** [comment]
@eileenmcnaughton I think the only sane thing to do is to drop the FK reference & deprecate the `cart_id` field.

**Comment by eileenmcnaughton** [comment]
Well it all worked on the demo site when I 

a) enabled the extension

```
    CRM.api3('Extension', 'enable', {
      "keys": "eventcart"
    }).then(function(result) {
      // do something with result
    }, function(error) {
      // oops
    });
```

b) enabled the setting under event settings

c) navigated to the cart via the event info page

EXCEPT for

the event template was only partially created - the reserved row but not the non-reserved row

**Comment by eileenmcnaughton** [comment]
OK - think that was the key was the same

**Comment by eileenmcnaughton** [comment]
@demeritcowboy I'm struggling to figure out the error handler test here - it starts at NULL & ends with a phpunit one that seems sane - I fixed the notice in the template but that didn't stop this failure

```

CRM_Financial_Form_PaymentFormsTest::testEventPaymentForms
Error handler is not the same at the end of the test as when it started. Start: 
End: PHPUnit\Util\ErrorHandler Object
(
    [convertDeprecationsToExceptions:PHPUnit\Util\ErrorHandler:private] => 1
    [convertErrorsToExceptions:PHPUnit\Util\ErrorHandler:private] => 1
    [convertNoticesToExceptions:PHPUnit\Util\ErrorHandler:private] => 1
    [convertWarningsToExceptions:PHPUnit\Util\ErrorHandler:private] => 1
    [registered:PHPUnit\Util\ErrorHandler:private] => 1
)

/home/homer/buildkit/build/build-3/web/sites/all/modules/civicrm/tests/phpunit/CiviTest/CiviUnitTestCase.php:575
/home/homer/buildkit/extern/phpunit9/phpunit9.phar:2307
```

**Comment by demeritcowboy** [comment]
If I remember right it shouldn't start at null, so maybe something about how the extension sets itself up for tests?

**Comment by demeritcowboy** [comment]
Ok this is a core test, that is testing extensions. Maybe installing an extension in a core test resets something about the error handler to null? I'm just guessing at the moment.

**Comment by demeritcowboy** [comment]
Oh setup doesn't call parent:setUp - that might be it.

**Comment by eileenmcnaughton** [comment]
ah @demeritcowboy its so obvious now you have pointed it out....

File: CRM/Upgrade/Incremental/php/FiveSeventySix.php [hunk 1]
```diff
@@ -57,4 +60,29 @@ public static function create_mesage_header_token() {
...
+   *
+   * It would be nice to delete the message template but who knows there could be a gotcha.
+   *
+   * @return bool
+   * @throws \CRM_Core_Exception
+   */
+  public static function removeEventCartAssets(): bool {
+    try {
+      CRM_Core_DAO::executeQuery("UPDATE civicrm_msg_template SET is_active = 0 WHERE workflow_name = 'event_registration_receipt'");
+      if (!CRM_Core_DAO::singleValueQuery('SELECT id FROM civicrm_events_in_carts LIMIT 1')) {
```
Comment (seamuslee001) [inline CRM/Upgrade/Incremental/php/FiveSeventySix.php:74]: @eileenmcnaughton is this duplicated in the below if ?

Comment (demeritcowboy) [reply CRM/Upgrade/Incremental/php/FiveSeventySix.php:74]: One is "carts" and one is "in carts".

Comment (seamuslee001) [reply CRM/Upgrade/Incremental/php/FiveSeventySix.php:74]: wonderful

**Comment by colemanw** [comment]
@eileenmcnaughton I removed the sql install/uninstall files in my commit and then you added them back, but they are redundant, at least the tables that are already declared in the schema/entityType.php files are...

**Comment by eileenmcnaughton** [comment]
@colemanw ok - I removed it - still seems to enable & work

**Comment by colemanw** [comment]
@eileenmcnaughton looking at this again and how it relates to the roadmap:
https://github.com/civicrm/civicrm-core/blob/85c643be0ac3341dd3a05ae5740f88676adbf21a/ext/eventcart/README.md#L9-L11

I think this PR was slightly premature. The vast majority of sites will have the extension enabled, because there's no way to disable it. We need to do this again, this time for people with the setting disabled, and uninstall the extension for them as well.

**Pull request merged**